db/migrate/20160317142959_add_index_to_switch.rb

```diff
@@ -0,0 +1,5 @@
+class AddIndexToSwitch < ActiveRecord::Migration[5.0]
+  def change
+    add_index :switches, :uid_ems, :unique => true
```

Comment:
After discussion with @agrare and @jrafanie, the lastest is using
`switch.uid_ems = ems.guid + host.ems_ref + switch.uid_ems`
  
A simpler/better solution would be using host.uuid + switch.uid_ems.  However, currently we don't collect host.uuid.